File: C# Excel Examples/RangesApiDemo.cs
using Spire.Cloud.Excel.Sdk.Client;
using Spire.Cloud.Excel.Sdk.Model;
using Spire.Cloud.Excel.Sdk.Api;
using System;
using System.Collections.Generic;
namespace CloudExcel
{
    class RangesApiDemo
    {
        static String appId = "your id";
        static String appKey = "your key";
        static String baseUrl = "https://api.e-iceblue.cn";
        static Configuration configuration = new Configuration(appId, appKey, baseUrl);
        static RangesApi rangesApi = new RangesApi(configuration);
        public static void CopyRanges()
        {
            string name = "CopyRanges.xlsx";
            string folder = "input";
            string storage = null;
            string sheetName = "Sheet1";
            RangeCopyRequest rangeOperate = new RangeCopyRequest();
            rangeOperate.Operate = "copystyle";
            Range range = new Range();
            range.FirstColumn = 1;
            range.FirstRow = 1;
            range.RowCount = 10;
            range.ColumnCount = 10;
            Range range2 = new Range();
            range2.FirstRow = 30;
            range2.FirstColumn = 1;
            range2.RowCount = 5;
            range2.ColumnCount = 8;
            rangeOperate.Source = range;
            rangeOperate.Target = range2;
            rangesApi.CopyRanges(name, sheetName, rangeOperate, folder, storage);
        }

        public static void GetRangeValue()
        {
            string name = "GetRangeValue.xlsx";
            string sheetName = "Sheet1";
            string folder = "input";
            string storage = null;
            string namerange = "C1:D3";
            var response = rangesApi.GetRangeValue(name, sheetName, namerange, null, null, null, null, folder, storage);
        }

        public static void MergeRange()
        {
            string name = "MergeRange.xlsx";
            string folder = "input";
            string storage = null;
            string sheetName = "Sheet1";
            Range range = new Range();
            range.FirstRow = 3;
            range.FirstColumn = 3;
            range.RowCount = 5;
            range.ColumnCount = 8;
            rangesApi.MergeRange(name, sheetName, range, folder, storage);
        }

        public static void MoveRange()
        {
            string name = "MoveRange.xlsx";
            string folder = "input";
            string storage = null;
            string sheetName = "Sheet1";
            Range range = new Range();
            range.FirstRow = 1;
            range.FirstColumn = 1;
            range.ColumnCount = 1;
            range.RowCount = 1;
            int? destRow = 25;
            int? destColumn = 3; 
            rangesApi.MoveRange(name, sheetName, range, destRow, destColumn, folder, storage);
        }
        public static void SetRangeColumnWidth()
        {
            string name = "SetRangeColumnWidth.xlsx";
            string folder = "input";
            string storage = null;
            string sheetName = "Sheet1";
            Range range = new Range();
            range.FirstRow = 1;
            range.FirstColumn = 1;
            range.ColumnCount = 5;
            range.RowCount = 2;
            double? value = 15;
            rangesApi.SetRangeColumnWidth(name, sheetName, range, value, folder, storage);
        }
        public static void SetRangeOutlineBorder()
        {
            string name = "SetRangeOutlineBorder.xlsx";
            string folder = "input";
            string storage = null;
            string sheetName = "Sheet1";
            RangeSetOutlineBorderRequest rangeOperate =
                new RangeSetOutlineBorderRequest();
            rangeOperate.BorderEdge = "outline";
            rangeOperate.BorderStyle = "Medium";
            rangeOperate.BorderColor = new Color(255, 255, 0, 0);
            Range range = new Range();
            range.FirstColumn = 1;
            range.FirstRow = 1;
            range.ColumnCount = 10;
            range.RowCount = 5;
            rangeOperate.Range = range;
            rangesApi.SetRangeOutlineBorder(name, sheetName, rangeOperate, folder, storage);
        }

        public static void SetRangeRowHeight()
        {
            string name = "SetRangeRowHeight.xlsx";
            string folder = "input";
            string storage = null;
            string sheetName = "Sheet1";
            Range range = new Range();
            range.FirstRow = 1;
            range.FirstColumn = 1;
            range.ColumnCount = 5;
            range.RowCount = 10;
            double? value = 20;
            rangesApi.SetRangeRowHeight(name, sheetName, range, value, folder, storage);
        }

        public static void SetRangeStyle()
        {
            string name = "SetRangeStyle_1.xlsx";
            string folder = "input";
            string storage = null;
            string sheetName = "Sheet1";
            RangeSetStyleRequest rangeOperate = new RangeSetStyleRequest();
            rangeOperate.Style = new Style();
            rangeOperate.Style.Font = new Font();
            rangeOperate.Style.Font.Color = new Color(255, 255, 0, 0);
            rangeOperate.Style.Font.Underline = "single";
            rangeOperate.Style.Font.Size = 12;
            rangeOperate.Style.Font.Name = "楷体";
            rangeOperate.Style.Font.IsItalic = true;
            rangeOperate.Style.Font.IsStrikeout = true;
            rangeOperate.Style.Font.IsBold = true;
            rangeOperate.Style.BorderCollection = new List<Border>();
            rangeOperate.Style.BorderCollection.Add(new Border("Medium", new Color(255, 255, 0, 0), "EdgeTop"));
            rangeOperate.Style.BorderCollection.Add(new Border("DashDot", new Color(255, 0, 255, 0), "EdgeRight"));
            rangeOperate.Style.BorderCollection.Add(new Border("Thin", new Color(0, 160, 32, 240), "EdgeLeft"));
            rangeOperate.Style.HorizontalAlignment = "left";
            rangeOperate.Style.IndentLevel = 1;
            rangeOperate.Style.TextDirection = "RightToLeft";
            rangeOperate.Style.BackgroundColor = new Color(0, 255, 255, 0);
            rangeOperate.Style.IsTextWrapped = true;
            Range range = new Range();
            range.FirstColumn = 2;
            range.FirstRow = 1;
            range.ColumnCount = 1;
            range.RowCount = 10;
            rangeOperate.Range = range;
            rangesApi.SetRangeStyle(name, sheetName, rangeOperate, folder, storage);
        }

        public static void SetRangeValue()
        {
            string name = "SetRangeValue.xlsx";
            string folder = "input";
            string storage = null;
            string sheetName = "Sheet2";
            Range range = new Range();
            range.FirstRow = 1;
            range.FirstColumn = 1;
            range.ColumnCount = 5;
            range.RowCount = 2;
            string value = "hello,冰蓝";
            bool? isConverted = true;
            bool? setStyle = false;
            rangesApi.SetRangeValue(name, sheetName, range, value, isConverted, setStyle, folder, storage);
        }

        public static void UnMergeRange()
        {
            string name = "UnMergeRange.xlsx";
            string folder = "input";
            string storage = null;
            string sheetName = "Sheet1";
            Range range = new Range();
            range.FirstRow = 3;
            range.FirstColumn = 4;
            range.RowCount = 2;
            range.ColumnCount = 5;
            rangesApi.UnMergeRange(name, sheetName, range, folder, storage);
        }
    }
}

File: Java Excel Examples/RangesApiDemo.java
import spire.cloud.excel.sdk.*;
import spire.cloud.excel.sdk.api.RangesApi;
import spire.cloud.excel.sdk.model.*;

import java.util.ArrayList;

public class RangesApiDemo {
    static String appId = "your id";
    static String appKey = "your key";
    static String baseUrl = "https://api.e-iceblue.cn";
    static Configuration configuration = new Configuration(appId, appKey, baseUrl);
    static RangesApi rangesApi = new RangesApi(configuration);

    public static void copyRanges() throws ApiException {
        String name = "CopyRanges.xlsx";
        String folder = "input";
        String storage = null;
        String sheetName = "Sheet1";
        RangeCopyRequest rangeOperate = new RangeCopyRequest();
        rangeOperate.setOperate("copystyle");
        Range range = new Range();
        range.firstColumn(1);
        range.firstRow(1);
        range.rowCount(10);
        range.columnCount(10);
        Range range2 = new Range();
        range2.firstColumn(30);
        range2.firstRow(1);
        range2.rowCount(5);
        range2.columnCount(8);
        rangeOperate.source(range);
        rangeOperate.target(range2);
        rangesApi.copyRanges(name, sheetName, rangeOperate, folder, storage);
    }

    public static void getRangeValue() throws ApiException {
        String name = "GetRangeValue.xlsx";
        String sheetName = "Sheet1";
        String folder = "input";
        String storage = null;
        String nameRange = "C1:D3";
        RangeValue response = rangesApi.getRangeValue(name, sheetName, nameRange, null, null, null, null, folder, storage);
    }

    public static void mergeRange() throws ApiException {
        String name = "MergeRange.xlsx";
        String folder = "input";
        String storage = null;
        String sheetName = "Sheet1";
        Range range = new Range();
        range.firstColumn(3);
        range.firstRow(3);
        range.rowCount(5);
        range.columnCount(8);
        rangesApi.mergeRange(name, sheetName, range, folder, storage);
    }

    public static void moveRange() throws ApiException {
        String name = "MoveRange.xlsx";
        String folder = "input";
        String storage = null;
        String sheetName = "Sheet1";
        Range range = new Range();
        range.firstColumn(1);
        range.firstRow(1);
        range.rowCount(1);
        range.columnCount(1);
        int destRow = 25;
        int destColumn = 3;
        rangesApi.moveRange(name, sheetName, range, destRow, destColumn, folder, storage);
    }
    public static void setRangeColumnWidth() throws ApiException {
        String name = "SetRangeColumnWidth.xlsx";
        String folder = "input";
        String storage = null;
        String sheetName = "Sheet1";
        Range range = new Range();
        range.firstColumn(1);
        range.firstRow(1);
        range.rowCount(2);
        range.columnCount(5);
        double value = 15;
        rangesApi.setRangeColumnWidth(name, sheetName, range, value, folder, storage);
    }
    public static void setRangeOutlineBorder() throws ApiException {
        String name = "SetRangeOutlineBorder.xlsx";
        String folder = "input";
        String storage = null;
        String sheetName = "Sheet1";
        RangeSetOutlineBorderRequest rangeOperate =
                new RangeSetOutlineBorderRequest();
        rangeOperate.borderEdge("outline");//Available value: vertical, horizontal, outline, edgeleft and so on.
        rangeOperate.borderStyle("Medium");//Available value: Medium, DashDot, Thin, Dotted, Hair, MediumDashDot, MediumDashDotDot and so on.
        rangeOperate.borderColor(new Color(255, 255, 0, 0));

        Range range = new Range();
        range.firstColumn(1);
        range.firstRow(1);
        range.columnCount(10);
        range.rowCount(5);
        rangeOperate.range(range);
        rangesApi.setRangeOutlineBorder(name, sheetName, rangeOperate, folder, storage);
    }

    public static void setRangeRowHeight() throws ApiException {
        String name = "SetRangeRowHeight.xlsx";
        String folder = "input";
        String storage = null;
        String sheetName = "Sheet1";
        Range range = new Range();
        range.firstColumn(1);
        range.firstRow(1);
        range.columnCount(5);
        range.rowCount(10);
        double value = 20;
        rangesApi.setRangeRowHeight(name, sheetName, range, value, folder, storage);
    }

    public static void setRangeStyle() throws ApiException {
        String name = "SetRangeStyle_1.xlsx";
        String folder = "input";
        String storage = null;
        String sheetName = "Sheet1";
        RangeSetStyleRequest rangeOperate = new RangeSetStyleRequest();
        rangeOperate.setStyle(new Style());
        rangeOperate.getStyle().font(new Font());
        rangeOperate.getStyle().getFont().color(new Color(255, 255, 0, 0));
        rangeOperate.getStyle().getFont().underline( "single");
        rangeOperate.getStyle().getFont().size(12);
        rangeOperate.getStyle().getFont().name("楷体");
        rangeOperate.getStyle().getFont().isItalic(true);
        rangeOperate.getStyle().getFont().isBold(true);

        ArrayList<Border> lists = new ArrayList<Border>();
        Border border = new Border();
        border.setLineStyle("Medium");
        border.setBorderType("EdgeTop");
        border.setColor(new Color(255, 255, 0, 0));
        lists.add(border);

        Border border1 = new Border();
        border1.setLineStyle("DashDot");
        border1.setBorderType("EdgeRight");
        border1.setColor(new Color(255, 0, 255, 0));
        lists.add(border1);

        Border border2 = new Border();
        border2.setLineStyle("Thin");
        border2.setBorderType("EdgeLeft");
        border2.setColor(new Color(0, 160, 32, 240));
        lists.add(border2);
        rangeOperate.getStyle().setBorderCollection(lists);
        rangeOperate.getStyle().horizontalAlignment("left");//需要设置HorizontalAlignment该属性方可生效
        rangeOperate.getStyle().indentLevel(1);
        rangeOperate.getStyle().textDirection("RightToLeft");
        rangeOperate.getStyle().backgroundColor(new Color(0, 255, 255, 0));
        rangeOperate.getStyle().isTextWrapped(true);

        Range range = new Range();
        range.firstColumn(2);
        range.firstRow(1);
        range.columnCount(1);
        range.rowCount(10);
        rangeOperate.range(range);
        rangesApi.setRangeStyle(name, sheetName, rangeOperate, folder, storage);
    }

    public static void setRangeValue() throws ApiException {
        String name = "SetRangeValue.xlsx";
        String folder = "input";
        String storage = null;
        String sheetName = "Sheet2";
        Range range = new Range();
        range.firstRow(1);
        range.firstColumn(1);
        range.columnCount(5);
        range.rowCount(2);
        String value = "E-iceblue";
        boolean isConverted = true;
        boolean setStyle = false;
        rangesApi.setRangeValue(name, sheetName, range, value, isConverted, setStyle, folder, storage);
    }

    public static void unmergeRange() throws ApiException {
        String name = "UnMergeRange.xlsx";
        String folder = "input";
        String storage = null;
        String sheetName = "Sheet1";
        Range range = new Range();
        range.firstColumn(3);
        range.firstRow(3);
        range.rowCount(5);
        range.columnCount(8);
        rangesApi.unmergeRange(name, sheetName, range, folder, storage);
    }
}
